### PR TITLE
Send "Key Pressed" events to native views

### PR DESCRIPTION
**Pull request by AlexanderFarkas**
Closes [issue#1937](https://github.com/leancodepl/patrol/issues/1937)

I think I need some guidance on how to write internal test for the feature.

### CODE REVIEW

**Comment by bartekpacia** [comment]
Please fix formatting.

> I think I need some guidance on how to write internal test for the feature.

You can add a new screen to the app found at `dev/e2e_app` and write a simple test that uses the new method you're adding.

File: packages/patrol/pubspec.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ homepage: https://patrol.leancode.co
 repository: https://github.com/leancodepl/patrol/tree/master/packages/patrol
 issue_tracker: https://github.com/leancodepl/patrol/issues
 documentation: https://patrol.leancode.co
-topics: [test, integration-test, ui-test, patrol]
+topics: [ test, integration-test, ui-test, patrol ]
```
Comment (bartekpacia) [inline packages/patrol/pubspec.yaml:10]: please revert these changes

File: schema.dart [hunk 2]
```diff
@@ -215,8 +221,7 @@ abstract class NativeAutomator<IOSServer, AndroidServer, DartClient> {
   void tapOnNotification(TapOnNotificationRequest request);
 
 // permissions
-  PermissionDialogVisibleResponse isPermissionDialogVisible(
-      PermissionDialogVisibleRequest request);
+  PermissionDialogVisibleResponse isPermissionDialogVisible(PermissionDialogVisibleRequest request);
```
Comment (bartekpacia) [inline schema.dart:224]: revert this change since it doesn't seem to be directly related to this PR

File: packages/patrol/lib/src/native/native_automator.dart [hunk 3]
```diff
@@ -66,11 +66,9 @@ class NativeAutomatorConfig {
       defaultValue: '8081',
     ),
     this.packageName = const String.fromEnvironment('PATROL_APP_PACKAGE_NAME'),
-    this.iosInstalledApps =
-        const String.fromEnvironment('PATROL_IOS_INSTALLED_APPS'),
+    this.iosInstalledApps = const String.fromEnvironment('PATROL_IOS_INSTALLED_APPS'),
     this.bundleId = const String.fromEnvironment('PATROL_APP_BUNDLE_ID'),
-    this.androidAppName =
-        const String.fromEnvironment('PATROL_ANDROID_APP_NAME'),
+    this.androidAppName = const String.fromEnvironment('PATROL_ANDROID_APP_NAME'),
```
Comment (bartekpacia) [inline packages/patrol/lib/src/native/native_automator.dart:71]: revert these changes since they doesn't seem to be directly related to this PR

**Comment by pdenert** [comment]
Hey @AlexanderFarkas are you still going to work on this?

**Comment by AlexanderFarkas** [comment]
@pdenert Unfortunately, now I don't have time to work on it. My team decided to wait for this package to stabilise a bit.
Given that internal structure was redone (unifying darwin implementation) after I wrote this PR, I believe this PR became obsolete.

**Comment by BishopSam** [comment]
@bartekpacia  

was this later implemented? cant find anything about key pressed events on the patrol docs

**Comment by piotruela** [comment]
@BishopSam Not yet. We will take care of finishing this
